### PR TITLE
Consolidate fl-tutorials dataset tooling into a shared datasets/ tree

### LINKED ISSUE #1069: Consolidate fl-tutorials dataset tooling into a shared datasets/ tree
### Description

The tutorial datasets' download/derive/enrich tooling is scattered across the two backend trees, and it is duplicating and drifting:

- **xray**: near-identical HF `snapshot_download` recipes in `fl-tutorials/nvflare/Makefile` and `fl-tutorials/flower/Makefile`, already drifted (different repo variable names, different output CSV filename), each downloading into its own `data/.hf` cache — testing both backends downloads the same data twice.
- **spleen**: the scripts (`download_spleen_dataset.py`, `create_spleen_accession_csv.py`, `upload_spleen_labels_to_xnat.py`) live under one tutorial's tree (`nvflare/image_segmentation/3d_spleen_segmentation/utils/`) but serve segmentation + evaluation + synthesis tutorials on both backends; the Flower Makefile delegates cross-tree with an apologetic comment.
- **synthea** (EHR risk prediction, #1068): second instance of the same delegate-to-the-other-backend pattern — `build_synthea_dataframe.py` is backend-agnostic but lives under `nvflare/tabular_classification/`.
- Dataset↔tutorial is many-to-many (spleen serves 3+ tutorials; arkplus serves fine-tuning + two evals), so per-tutorial `utils/` is the wrong home.

Consolidate into a shared `fl-tutorials/datasets/<name>/` tree (xray, spleen, arkplus; synthea follows in #1068), each dataset owning its download/derive/enrich scripts and Make targets, with a single gitignored shared data output root so both backends reuse one download.

### Acceptance criteria

- [x] `fl-tutorials/datasets/<name>/` owns each dataset's scripts + Make targets; no dataset scripts remain under per-tutorial `utils/` and no duplicated download recipes remain in the per-backend Makefiles.
- [x] Existing entry points keep working unchanged: `make -C fl-tutorials download-xray-data|download-spleen-data|download-arkplus-*|upload-spleen-labels [FL_BACKEND=flower]`.
- [x] One shared gitignored data root — downloading a dataset once serves both backends' tutorials (per-backend variants, e.g. the Flower pre-built FLIP-format spleen tree vs the NVFLARE MSD build, remain available as distinct targets/outputs).
- [x] All consumers re-pointed: per-tutorial `.env.app` paths, `flower/run-tutorial.sh` mappings, flip-api `e2e_smoke_spleen*` defaults, root `CLAUDE.md`/`AGENTS.md`, READMEs, docs.
- [x] `make -C fl-tutorials test` green; tutorial-sync CI green; downloads verified to land in the layout the tutorials expect.

### Technical considerations

- The `fl-apps` upload bundles must stay self-contained — dataset tooling is harness-side only and must not move anything out of `app_files/`/`app/`.
- `upload_spleen_labels_to_xnat.py` is referenced by root `CLAUDE.md`, flip-api Makefile (`ENRICHMENT_ARGS` targets) and docs; all references move in the same PR.
- #1068 (EHR tutorial) lands `datasets/synthea/` on top of this structure after it merges.

### PR DESCRIPTION
**Pull request by atriaybagur**
## What

Consolidates the tutorial datasets' download/derive/enrich tooling into a shared **`fl-tutorials/datasets/`** tree, with all outputs in one shared gitignored **`fl-tutorials/data/`** root. Closes #1069.

Datasets are backend-agnostic (a tutorial's NVFLARE and Flower twins train on the same data) and many-to-one (spleen alone backs segmentation, evaluation and diffusion), but the tooling was scattered per backend and drifting: near-identical xray recipes in both backend Makefiles (different output CSV names, separate `.hf` caches — testing both backends downloaded the same data twice), and the spleen scripts living under one tutorial's `utils/` with the Flower Makefile delegating cross-tree behind an apologetic comment.

## How

- **`fl-tutorials/datasets/Makefile`** owns every dataset target (single copy): `download-xray-data`, `download-spleen-data` (FL_BACKEND-dispatched to the explicit `download-spleen-msd-data` / `download-spleen-flip-format-data` variants), `download-arkplus-{finetuning,eval}-data`, `upload-spleen-labels`. The fl-tutorials root Makefile forwards these there, so **every documented entrypoint is unchanged** (`make -C fl-tutorials download-spleen-data NUM_CASES=41 [FL_BACKEND=flower]`, etc.).
- **`fl-tutorials/datasets/spleen/`** takes the three spleen scripts and their uv project (moved from `nvflare/image_segmentation/3d_spleen_segmentation/{utils/,pyproject.toml}`; the project env existed solely for these utilities), plus a new `download_spleen_flip_format_dataset.py` for the FLIP-format variant. `datasets/README.md` documents each dataset's source, layout and consumers.
- **Every download is a script, not an inline `python -c`**: the xray, arkplus and FLIP-format spleen fetches each live in their dataset folder (`xrays_mini_300/`, `arkplus/`, `spleen/`) and run via `uv run --no-project --with huggingface_hub`; the Makefile targets just invoke them.
- **One shared data root**: both backends now read `fl-tutorials/data/`; one download serves both. The xray CSV gets one canonical name (`dataframe.csv`); the two spleen variants (MSD build: `images/` + `dataframe.csv`; FLIP-format: `accession-resources/` + `sample_get_dataframe_response.csv` + checkpoint) coexist under `data/spleen/` — the FLIP-format download now removes only its own outputs instead of `rm -rf data/spleen`.
- **Consumers re-pointed**: the 7 nvflare `.env.app` files, `flower/run-tutorial.sh` (which now also fail-fasts on a missing dataframe file, not just the images dir), flip-api's `e2e_smoke_spleen*` enrichment wiring (`SPLEEN_ENRICHMENT_CWD`/`CMD`/`LABELS_DIR`), root `CLAUDE.md`/`AGENTS.md`, tutorial READMEs, `fl-services/flower/README.md`, the demo-video docs and the arkplus user guide. The upload script's remediation text now names the two *variants* instead of the two backends.
- Removes the accidentally committed self-referencing `fl-tutorials/nvflare/data` symlink.
- Strips the leftover Flower example-gallery frontmatter (`tags`/`dataset`/`framework`) from the two Flower spleen tutorial READMEs — it is metadata for Flower's examples website and renders on GitHub as a stray horizontal rule plus literal text.
- A side-effect fix: the `create-flip-app-from-flower.rst` local-testing example (`../../data/spleen/...` from the tutorial root) pointed at a path that didn't exist under the old layout — it is exactly right under the new one.

## Verification

- `make -C fl-tutorials download-xray-data` and `make -C fl-tutorials download-spleen-data FL_BACKEND=flower` run end-to-end into the shared root, with the expected layouts, and `git status` stays clean afterwards (gitignore covers the new root).
- Target dispatch, `NUM_CASES` forwarding, and the per-variant `SPLEEN_LABELS_DIR` defaults verified via `make -n` command expansion; `bash -n` on `run-tutorial.sh`; `list-tutorials` works on both backends.
- `make -C fl-tutorials test` green (ruff + 112 passed, 12 expected skips — the count grew when `develop` was merged in: its new FLIP#1060 spleen case-selection tests were re-pointed at the moved `datasets/spleen/` downloader as part of the merge).
- `scripts/check_tutorial_sync.sh` green; a repo-wide grep finds zero references to the old paths.
- Not run (large downloads, unchanged recipes beyond path prefixes): the MSD spleen and arkplus downloads; a live `upload-spleen-labels` against XNAT.

## Sequencing

**Merge before #1068** — the EHR risk-prediction tutorial then rebases onto this and moves its `build_synthea_dataframe.py` into `datasets/synthea/`, instead of adding a second cross-tree delegation.

## Acceptance Criteria

*Imported from issue #1069*

- [x] `fl-tutorials/datasets/<name>/` owns each dataset's scripts + Make targets; no dataset scripts remain under per-tutorial `utils/` and no duplicated download recipes remain in the per-backend Makefiles.
- [x] Existing entry points keep working unchanged: `make -C fl-tutorials download-xray-data|download-spleen-data|download-arkplus-*|upload-spleen-labels [FL_BACKEND=flower]`.
- [x] One shared gitignored data root — downloading a dataset once serves both backends' tutorials (per-backend variants, e.g. the Flower pre-built FLIP-format spleen tree vs the NVFLARE MSD build, remain available as distinct targets/outputs).
- [x] All consumers re-pointed: per-tutorial `.env.app` paths, `flower/run-tutorial.sh` mappings, flip-api `e2e_smoke_spleen*` defaults, root `CLAUDE.md`/`AGENTS.md`, READMEs, docs.
- [x] `make -C fl-tutorials test` green; tutorial-sync CI green; downloads verified to land in the layout the tutorials expect.

### CODE REVIEW

**Comment by atriaybagur** [comment]
Ready for review. develop merged in (only conflict: the CLAUDE.md/AGENTS.md repo-structure line vs the new map-apps/ entry; also re-pointed the new #1060 spleen case-selection test at the moved datasets/spleen/ downloader — the natsort fix itself carried through the rename). All 34 checks green on the merged head, incl. tutorial-sync and the full Cypress/heavy suites; `make -C fl-tutorials test` locally: ruff clean, 112 passed / 12 expected skips. Copilot review requested. Note the sequencing section: this merges before #1068.

File: fl-tutorials/datasets/Makefile [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+	@$(SPLEEN_UV) python spleen/download_spleen_dataset.py --output_dir $(DATA_DIR)/spleen/images --num_cases $(NUM_CASES)
+	@$(SPLEEN_UV) python spleen/create_spleen_accession_csv.py --images_dir $(DATA_DIR)/spleen/images --output_csv $(DATA_DIR)/spleen/dataframe.csv
+	@echo "✅ MSD spleen data ready at $(abspath $(DATA_DIR))/spleen/ — make -C fl-tutorials run-tutorial TUTORIAL=3d_spleen_segmentation"
+
+# Download the pre-built FLIP-format spleen tree (fixed 6-case snapshot; ignores NUM_CASES)
+# plus the evaluation-tutorial checkpoint. Removes only its own outputs, so an MSD build in
+# the same data/spleen/ survives a re-download.
+download-spleen-flip-format-data:
+	@echo "⬇️  Downloading FLIP-format spleen dataset from Hugging Face ($(HF_TEST_DATA_REPO))..."
+	@uv run --no-project --with huggingface_hub python -c "from huggingface_hub import snapshot_download; snapshot_download(repo_id='$(HF_TEST_DATA_REPO)', repo_type='dataset', local_dir='$(HF_CACHE)', allow_patterns=['flip-fl-base-test-data/spleen/*', 'flip-fl-base-test-data/checkpoints/*'])"
```
Comment (garciadias) [inline fl-tutorials/datasets/Makefile]: This target (download-spleen-flip-format-data) is the last of the three dataset downloads still using an inline uv run --with huggingface_hub python -c one-liner. The sibling commit in this PR (Pythonise the xray and arkplus dataset downloads) moved exactly this pattern into dedicated scripts for xray and arkplus for readability/testability -- this one is arguably the best candidate for the same treatment since it does the most work (three copies + a checkpoint) of the three. Non-blocking.

**Review by garciadias — DISMISSED** [review]
Clean, mechanical, thoroughly self-verified refactor consolidating dataset tooling; all path rewrites across .env.app files, Makefiles, run-tutorial.sh, flip-api/Makefile, and the CLAUDE.md/AGENTS.md twin-file pair check out, no stale references remain, CI is green. One optional non-blocking nit left inline. Approving.

File: fl-tutorials/datasets/Makefile (hunk 1 above)
Comment (atriaybagur) [reply fl-tutorials/datasets/Makefile]: Done in 615d0973: `datasets/spleen/download_spleen_flip_format_dataset.py` now carries the snapshot_download + the three copies (accession-resources/, sample_get_dataframe_response.csv, the evaluation checkpoint into data/model_checkpoints/model.pt), on the same argparse/function shape as the xray and arkplus scripts; the Makefile target is reduced to invoking it. It stays a `uv run --no-project --with huggingface_hub` call rather than adopting spleen/'s uv project — that env exists for the MSD scripts' MONAI/torch stack, which a pure Hugging Face fetch needs none of (the target comment and datasets/README.md say so). It still replaces only its own outputs, so an MSD build beside them in data/spleen/ survives a re-download — verified by running `make -C fl-tutorials download-spleen-data FL_BACKEND=flower` twice against a probe images/ + dataframe.csv, which came through intact; `make -C fl-tutorials test` green (112 passed, 12 skipped).
